common/data_format.hpp

```diff
@@ -174,17 +174,68 @@ namespace ILLIXR {
 		time_point predict_target_time; // Time that prediction targeted.
 	} fast_pose_type;
 
+	// Used to identify which graphics API is being used (for swapchain construction)
+	enum class graphics_api {
+		OPENGL,
+		VULKAN
+	};
+
+	typedef struct vk_image_handle {
+		int file_descriptor;
+		GLuint64 format;
+		GLuint64 allocation_size;
+		GLsizei width, height;
+
+		vk_image_handle(int fd_, GLuint64 format_, GLuint64 alloc_size, GLsizei width_, GLsizei height_)
+			: file_descriptor{fd_}
+			, format{format_}
+			, allocation_size{alloc_size}
+			, width{width_}
+			, height{height_}
+		{}
+	} vk_image_handle;
+
+	// This is used to share swapchain images between ILLIXR and Monado.
+	// When Monado uses its GL pipeline, it's enough to just share a context during creation.
+	// Otherwise, file descriptors are needed to share the images.
+	struct image_handle : public switchboard::event {
+		graphics_api type;
+		union {
+			GLuint gl_handle;
+			vk_image_handle vk_handle;
+		};
+		// is there a better way to do this? All it does is store 
```

Comment:
In the future, perhaps we can have Monado publish all the swapchain information on a different topic. That way, we won't have to publish static information each time. But I don't see the current implementation causing any performance issues, so this is good for now!